pennylane/qchem/givens_decomposition.py

```diff
@@ -147,43 +147,68 @@ def givens_decomposition(U):
                             # Update U = T(N+j-i-1, N+j-i) @ U
 
     """
+    interface = qml.math.get_deep_interface(unitary)
+    unitary = qml.math.copy(unitary) if interface == "jax" else qml.math.toarray(unitary).copy()
```

Comment:
`qml.math.copy` is not general enough for all interfaces?

Reply:
No, it is not. 😅